crates/polars-core/src/chunked_array/mod.rs

```diff
@@ -442,6 +442,35 @@ impl AsSinglePtr for BinaryChunked {}
 #[cfg(feature = "object")]
 impl<T: PolarsObject> AsSinglePtr for ObjectChunked<T> {}
 
+pub enum ChunkedArrayLayout<'a, T: PolarsDataType> {
+    SingleNoNull(&'a T::Array),
```

Comment:
Seems redundant with `Single`. Maybe we should name this `SingleSlice` and get a slice here?

Reply:
No, it's supposed to indicate that there's no nulls in the array so you can use `get_value` and such instead of `get`. Most places that check for the layout also want to distinguish between nulls vs no nulls.

Reply:
Oh yeah. Right, that will boil down to a `get` on a slice if the backing array is numeric. All good. :+1:

Reply:
And if you don't want to distinguish you can simply match on `Single(a) | SingleNoNull(a)` and combine the cases.